packages/sitecore-jss-nextjs/src/middleware/redirects-middleware.ts

```diff
@@ -68,39 +58,33 @@ export class RedirectsMiddleware {
     return this.handler;
   }
 
-  protected excludeRoute(pathname: string) {
-    if (
-      pathname.includes('.') || // Ignore files
-      pathname.startsWith('/api/') || // Ignore Next.js API calls
-      pathname.startsWith('/sitecore/') || // Ignore Sitecore API calls
-      pathname.startsWith('/_next') // Ignore next service calls
-    ) {
-      return true;
-    }
-    return false;
-  }
-
-  protected getHostname(req: NextRequest) {
-    const hostHeader = req.headers.get('host')?.split(':')[0];
-    return hostHeader || this.defaultHostname;
-  }
-
   private handler = async (req: NextRequest, res?: NextResponse): Promise<NextResponse> => {
-    const hostname = this.getHostname(req);
-    const siteName = res?.cookies.get('sc_site')?.value || this.config.getSite(hostname).name;
+    const hostHeader = this.getHostHeader(req);
+    const hostname = hostHeader || this.defaultHostname;
+    const siteName =
+      res?.cookies.get(this.SITE_SYMBOL)?.value || this.config.getSite(hostname).name;
```

Comment:
I see we have similar site resolution logic (using cookie, host header, etc) in all middleware now. If we had `siteResolver` on base config, we could potentially also add a helper to base class to `getSite`. Not sure if it makes sense, but something to consider.

Reply:
Makes sense to me. Tried to extract that logic